nexus/src/app/background/sync_switch_port_settings.rs

```diff
@@ -0,0 +1,1236 @@
+                    generation: 0,
+                    schema_version: 1,
+                    body: EarlyNetworkConfigBody {
+                        ntp_servers: Vec::new(), //TODO
+                        rack_network_config: Some(RackNetworkConfigV1 {
+                            rack_subnet: subnet,
+                            //TODO(ry) you are here. We need to remove these too. They are
+                            // inconsistent with a generic set of addresses on ports.
+                            infra_ip_first: Ipv4Addr::UNSPECIFIED,
+                            infra_ip_last: Ipv4Addr::UNSPECIFIED,
+                            ports,
+                            bgp,
+                        }),
+                    },
+                };
+
+                // check to see if our config is different(?)
+                // we currently are not caching the last sent bootstore config, so we have to build
+                // it every time and compare.
+
+                let bootstore_needs_update = {
+                    match downstream_config {
+                        Some(ref existing_config) => {
+                            existing_config.schema_version != upstream_config.schema_version ||
+                            existing_config.body.ntp_servers != upstream_config.body.ntp_servers ||
+                            existing_config.body.rack_network_config != upstream_config.body.rack_network_config
+                        },
+                        _ => true,
+                    }
+                };
+
+                if bootstore_needs_update {
+                    let generation = match self.datastore
+                        .bump_bootstore_generation(opctx, NETWORK_KEY.into())
+                        .await {
+                        Ok(value) => value,
+                            Err(e) => {
+                                error!(
+                                    log,
+                                    "error while fetching next bootstore generation from db";
+                                    "key" => %NETWORK_KEY,
+                                    "error" => %e,
+                                );
+                                continue;
+                            },
+                        };
+
+                    upstream_config.generation = generation as u64;
+                    info!(
+                        &log,
+                        "updating bootstore config";
+                        "old config" => ?downstream_config,
+                        "new config" => ?upstream_config,
+                    );
+
+                    // push the updates to both scrimlets
+                    // if both scrimlets are down, bootstore updates aren't happening anyway
+                    for (location, client) in &sled_agent_clients {
+                        if let Err(e) = client.write_network_bootstore_config(&upstream_config).await {
```

Comment:
Maybe this can be relaxed to have at least one successful write semantics. I think if we write the bootstore to one sled agent, does not matter which, it's the bootstore protocol's responsibility to propagate to others. cc: @andrewjstone.

Reply:
I believe this is the relaxed implementation? If any (or all) of the requests fail, we simply log the error and continue. We try both scrimlets just in case one is down for a prolonged period of time.

Reply:
I think I better understand what you're saying, which is if we're successful on one, we don't need to try the second. I'm ok with that, but I think the eagerly updating multiple agents falls into the bootstore "seeding" idea that has been discussed offline, where the propagation happens faster if multiple agents receive the update instead of one, but I also defer to @andrewjstone on this.

Reply:
Yes, this is what I called "seeding". It's a mechanism used by some other protocols including [rapid](https://research.vmware.com/publications/stable-and-consistent-membership-at-scale-with-rapid). However, it's really just for reliability here and not for performance. If the original node dies permanently the data won't replicate. You get better odds with N replicas that such a thing won't happen.